### PR TITLE
add supporting remove partitions if delta was partitioned

### PR DESCRIPTION
**Pull request by KevinKarlBob**
It is PR for issue with removing partitions in delta table. 
It was discussed here https://delta-users.slack.com/archives/CJ70UCSHM/p1587048581235000
And i implemened proposed solution https://delta-users.slack.com/archives/CJ70UCSHM/p1587068793244400

### CODE REVIEW

File: src/main/scala/org/apache/spark/sql/delta/schema/ImplicitMetadataOperation.scala [hunk 1]
```diff
@@ -79,8 +79,7 @@ trait ImplicitMetadataOperation extends DeltaLogging {
     def isNewSchema: Boolean = txn.metadata.schema != mergedSchema
     // We need to make sure that the partitioning order and naming is consistent
     // if provided. Otherwise we follow existing partitioning
-    def isNewPartitioning: Boolean = normalizedPartitionCols.nonEmpty &&
-      txn.metadata.partitionColumns != normalizedPartitionCols
+    def isNewPartitioning: Boolean = txn.metadata.partitionColumns != normalizedPartitionCols
```
Comment (brkyvz) [inline src/main/scala/org/apache/spark/sql/delta/schema/ImplicitMetadataOperation.scala:82]: you need a more intricate condition here:
 1. `normalizedPartitionCols.nonEmpty && txn.metadata.partitionColumns != normalizedPartitionCols` OR
 2. `overwriteSchema &&  txn.metadata.partitionColumns != normalizedPartitionCols`

File: src/test/scala/org/apache/spark/sql/delta/DeltaSuite.scala [hunk 2]
```diff
@@ -494,6 +494,29 @@ class DeltaSuite extends QueryTest
...
+      if (tempDir.exists()) {
+        assert(tempDir.delete())
+      }
+
+      spark.range(100).select('id, 'id % 4 as 'by4)
+        .write
+        .format("delta")
+        .partitionBy("by4")
+        .save(tempDir.toString)
+
```
Comment (brkyvz) [inline src/test/scala/org/apache/spark/sql/delta/DeltaSuite.scala:508]: can you also add a check that the partitioning column indeed did exist

Comment (KevinKarlBob) [reply src/test/scala/org/apache/spark/sql/delta/DeltaSuite.scala:508]: check was added

File: src/main/scala/org/apache/spark/sql/delta/schema/ImplicitMetadataOperation.scala (hunk 1 above)
Comment (KevinKarlBob) [reply src/main/scala/org/apache/spark/sql/delta/schema/ImplicitMetadataOperation.scala]: i indroduced new variable for it

**Pull request closed without merging**